app/src/main/java/com/github/sdpteam15/polyevents/fragments/ProfileFragment.kt

```diff
@@ -1,32 +1,56 @@
 package com.github.sdpteam15.polyevents.fragments
 
+import android.app.Activity
 import android.os.Bundle
+import android.util.Log
 import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Button
+import android.widget.TextView
+import com.github.sdpteam15.polyevents.MainActivity
 import com.github.sdpteam15.polyevents.R
-
-// TODO: Rename parameter arguments, choose names that match
-// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
-private const val ARG_PARAM1 = "param1"
-private const val ARG_PARAM2 = "param2"
+import com.github.sdpteam15.polyevents.helper.HelperFunctions.changeFragment
+import com.github.sdpteam15.polyevents.user.UserInterface
+import com.github.sdpteam15.polyevents.user.User
+import com.google.firebase.auth.FirebaseAuth
 
 /**
- * A simple [Fragment] subclass.
- * Use the [ProfileFragment.newInstance] factory method to
- * create an instance of this fragment.
+ *  [Fragment] subclass that represents the profile page.
  */
-class ProfileFragment : Fragment() {
+class ProfileFragment : Fragment(){
+    //User that we can set manually for testing
+    private var testUser: UserInterface?=null
+    //Allow us to use a fake user for the tests
+    var currentUser: UserInterface?
+        get(){
+            if( testUser!= null) {return testUser}
+            else {return User.CurrentUser}
+        }
+        set(value){
+            testUser = value
+        }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        //If the user is not logged in, redirect him to the login page
+        if(currentUser == null){
+            changeFragment(activity, MainActivity.fragments[R.id.ic_login])
+        }
     }
 
-    override fun onCreateView(
-        inflater: LayoutInflater, container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_profile, container, false)
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        val viewRoot = inflater.inflate(R.layout.fragment_profile, container, false)
+        viewRoot.findViewById<Button>(R.id.btnLogout).setOnClickListener { _ ->
+            FirebaseAuth.getInstance().signOut()
```

Comment:
For next weeks : maybe we will need to consider to add a level of abstraction over the authentication via Firebase (cf discord " In particular, there shouldn't be any direct proprietary API calls in your business or view logic.").